Library/Homebrew/compilers.rb

```diff
@@ -105,11 +105,19 @@ def compiler
 
   private
 
+  def gnu_gcc_versions
+    # prioritize gcc version provided by gcc formula.
+    v = Formulary.factory("gcc").version.to_s.slice(/\d/)
+    GNU_GCC_VERSIONS - [v] + [v]
```

Comment:
It's not obvious here why the same thing is being added and removed; could this perhaps have a comment saying that it's being deliberately added to the end/beginning of the list specifically? Thanks!